smp/mptables.c

```diff
@@ -210,8 +210,9 @@ static void process_mpc_entries(mpc_hdr_t *mpc_ptr) {
             percpu_t *percpu = get_percpu_page(nr_cpus++);
 
             percpu->apic_id = mpc_cpu->lapic_id;
-            percpu->enabled = mpc_cpu->en;
-            percpu->bsp = mpc_cpu->bsp;
+            percpu->enabled = !!mpc_cpu->en;
+            percpu->bsp = !!mpc_cpu->bsp;
+            set_bsp_cpu_id(percpu->cpu_id);
```

Comment:
We do expect more than one MPC_PROCESSOR_ENTRY, right?
Doesn't this overwrite bsp_cpu_id global?

Reply:
Oh yes! It must be conditional of course. Thanks